cpp/src/glm/ols.cuh

```diff
@@ -62,7 +67,8 @@ void olsFit(const raft::handle_t& handle,
             bool fit_intercept,
             bool normalize,
             cudaStream_t stream,
-            int algo = 0)
+            int algo              = 0,
+            math_t* sample_weight = nullptr)
```

Comment:
Not something you need to change in this PR, but it would be nice to start adopting `std::optional` for arguments like these. Becomes self-documenting as well.

Reply:
I agree so I started to make the change but I saw that `std::optional` support on Cython is very recent and will be added to version `0.30`. For the moment we're using version `0.29` so the adoption of `std::optional` on Python-facing functions should maybe wait a bit more.